tests/posix/fs/src/test_fs_dir.c

```diff
@@ -88,16 +88,33 @@ static int test_lsdir(const char *path)
 	return res;
 }
 
+static void before_fn(void *unused)
+{
+	ARG_UNUSED(unused);
+
+	test_mkdir();
+}
+
+static void after_fn(void *unused)
+{
+	ARG_UNUSED(unused);
+
+	unlink(TEST_DIR_FILE);
+	unlink(TEST_DIR);
+}
+
+ZTEST_SUITE(posix_fs_dir_test, NULL, test_mount, before_fn, after_fn,
+	    test_unmount);
+
 /**
  * @brief Test for POSIX mkdir API
  *
  * @details Test creates a new directory through POSIX
  * mkdir API and open a new file under the directory and
  * writes some data into the file.
  */
-void test_fs_mkdir(void)
+ZTEST(posix_fs_dir_test, test_fs_mkdir)
 {
-	zassert_true(test_mkdir() == TC_PASS, NULL);
```

Comment:
Same here, seems that this is trying to make sure that the before function ran correctly. I would move the assert away from the before and into the test function. Can we check the side effects of the mkdir? Looking at `test_mkdir()` I would have this function assert that the directory exists.

Reply:
I'll add an extra check here, and add an `__ASSERT_NO_MSG()` until there is some means of asserting in setup / teardown.